client/src/components/classesPage/ClassEditView.js

```diff
@@ -1,19 +1,166 @@
-import React from "react";
-import Grid from "@material-ui/core/Grid";
-import { withStyles } from "@material-ui/core/styles";
+import React, { useState, useEffect } from 'react';
+import {
+  Grid,
+  Checkbox,
+  Card,
+  Select,
+  MenuItem,
+  Button
+} from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
 
 const styles = theme => ({
   wrapper: {
+    color: 'white'
+  },
+  cardList: {
+    display: 'flex'
+  },
+  card: {
+    width: 200,
+    height: 200,
+    border: '1px solid white',
+    margin: theme.spacing.unit * 3
   }
 });
 
 function ClassEditView(props) {
+  const { classes } = props;
+  const classId = props.match.params.id;
+  const ax = axios.create({
+    baseURL: 'http://localhost:9000' // development
+  });
+  const [students, setStudents] = useState([]);
+  const [refreshrs, setRefreshrs] = useState([]);
+  const [teacherRefs, setTeacherRefs] = useState([]);
+  const [selectedStudents, setSelectedStudents] = useState([]);
+  // const [classDetails, setClassDetails] = useState([]);
+
+  // get class details on mount
+  useEffect(() => {
+    fetchStudents();
+    fetchRefreshrs();
+    fetchTeacherRefreshrs();
+  }, []);
+
+  useEffect(() => {
+    console.log('students:', students);
+  }, [students]);
+
+  useEffect(() => {
+    console.log('refreshrs:', refreshrs);
+  }, [refreshrs]);
+
+  useEffect(() => {
+    console.log('teacherRefs:', teacherRefs);
+  }, [teacherRefs]);
+
+  useEffect(() => {
+    console.log('selectedStudents:', selectedStudents);
+  }, [selectedStudents]);
+
+  async function fetchStudents() {
+    console.log(classId);
+    const res = await ax.get(`/classes/${classId}/students`);
+    console.log(res);
+    setStudents(res.data);
+  }
+
+  async function fetchRefreshrs() {
+    const res = await ax.get(`/classes/${classId}/refreshrs`);
+    console.log(res);
+    setRefreshrs(res.data);
+  }
+
+  async function fetchTeacherRefreshrs(id) {
+    // this should be user id, not 35
+    const res = await ax.get('/refreshrs/teachers/35');
+    const unassignedRefreshrs = res.data.filter(r => !refreshrs.includes(r)); // unsure if this filter will work, need to test
```

Comment:
Looks like it should work, the filter.

Reply:
Yeah, I think so, just didn't have test data this morning to run it.